p2p/node/node.go

```diff
@@ -234,20 +235,21 @@ func (p *P2PNode) p2pAddress() (multiaddr.Multiaddr, error) {
 // Helper to access the corresponding data cache
 func (p *P2PNode) pickCache(datatype interface{}, location common.Location) *lru.Cache[common.Hash, interface{}] {
 	switch datatype.(type) {
-	case *types.WorkObject:
+	case *types.WorkObject, *types.WorkObjectHeaderView, *types.WorkObjectBlockView:
 		return p.cache[location.Name()]["blocks"]
 	case *types.Transaction:
 		return p.cache[location.Name()]["transactions"]
 	case *types.Header:
 		return p.cache[location.Name()]["headers"]
 	default:
-		log.Global.Fatalf("unsupported type")
+		log.Global.WithField("type", reflect.TypeOf(datatype)).Fatalf("unsupported type")
 		return nil
 	}
 }
 
 // Add a datagram into the corresponding cache
 func (p *P2PNode) cacheAdd(hash common.Hash, data interface{}, location common.Location) {
+	return
```

Comment:
will make issue to get the cache to work

Reply:
https://github.com/dominant-strategies/go-quai/issues/1760